### PR TITLE
Add code examples for `torch_geometric.utils` functions

### PR DESCRIPTION
**Pull request by EdisonLeeeee**
This PR is to add some code examples to doc-strings for `torch_geometric.utils` functions, as mentioned in [this issue](https://github.com/pyg-team/pytorch_geometric/issues/5383). Currently, the following functions imported in `torch_geometric.utils.__init__` are listed (ordered by the filenames):

- [x] **coalesce.py**: `coalesce`
- [x] **convert.py**: `to_scipy_sparse_matrix`, `from_scipy_sparse_matrix`
- [x] **convert.py**: `to_networkx`, `from_networkx`
- [x] **convert.py**: `to_trimesh`, `from_trimesh`
- [ ] **convert.py**: `to_cugraph`, `from_cugraph`
- [x] **degree.py**: `degree`
- [x] **dropout.py**: `dropout_adj`
- [x] **geodesic.py**: `geodesic_distance`
- [x] **get_laplacian.py**: `get_laplacian`
- [ ] **get_mesh_laplacian.py**: `get_mesh_laplacian`
- [x] **grid.py**: `grid`
- [x] **homophily.py**: `homophily`
- [x] **isolated.py**: `contains_isolated_nodes`, `remove_isolated_nodes`
- [x] **loop.py**: `contains_self_loops`, `remove_self_loops`,`segregate_self_loops`, `add_self_loops`,  `add_remaining_self_loops`, `get_self_loop_attr`
- [x] **mask.py**: `index_to_mask`, `mask_to_index`
- [x] **negative_sampling.py**: `negative_sampling`, `batched_negative_sampling`, `structured_negative_sampling`, `structured_negative_sampling_feasible`
- [x] **normalized_cut.py**: `normalized_cut`
- [x] **random.py**: `erdos_renyi_graph`, `stochastic_blockmodel_graph`, `barabasi_albert_graph`
- [ ] **scatter.py** `scatter`
- [ ] **smiles.py**: `from_smiles`
- [x] **softmax.py**: `softmax`
- [x] **sort_edge_index.py**: `sort_edge_index`
- [x] **sparse.py**: `dense_to_sparse`
- [x] **subgraph.py**: `get_num_hops`, `subgraph`, `k_hop_subgraph`, `bipartite_subgraph`
- [x] **to_dense_adj.py**: `to_dense_adj`
- [x] **to_dense_batch.py**: `to_dense_batch`
- [ ] ~~**train_test_split_edges.py**: `train_test_split_edges`~~(deprecated)
- [ ] **tree_decomposition.py**: `tree_decomposition`
- [x] **unbatch.py**: `unbatch`, `unbatch_edge_index`
- [x] **undirected.py**: `is_undirected`, `to_undirected`

Just wanted to let you know how this PR is doing now. I'll finish the remaining functions in the next few days. Please let me know if there is something missing or wrong.

### CODE REVIEW

File: torch_geometric/utils/coalesce.py [hunk 1]
```diff
@@ -37,6 +37,31 @@ def coalesce(
 
     :rtype: :class:`LongTensor` if :attr:`edge_attr` is :obj:`None`, else
         (:class:`LongTensor`, :obj:`Tensor` or :obj:`List[Tensor]]`)
+
+    Examples:
```
Comment (rusty1s) [inline torch_geometric/utils/coalesce.py:41]:
```suggestion
    Example:
```

File: torch_geometric/utils/coalesce.py [hunk 2]
```diff
@@ -37,6 +37,31 @@ def coalesce(
...
+
+        >>> edge_index = torch.tensor([[1, 1, 2, 3], [3, 3, 1, 2]])
```
Comment (rusty1s) [inline torch_geometric/utils/coalesce.py:43]: Can we move this to a `.. code-block:: python`block? I also feel it might be more straightforward to copy/paste if we skip the `>>>`. Might be also good to add the necessary imports.

Comment (EdisonLeeeee) [reply torch_geometric/utils/coalesce.py:43]: But we cannot display the outputs if we use a `.. code-block:: python` block. Do you think this is needed?

For the imports, I have made all the functions used from the `torch_geometric.utils` and `torch` modules only.

Comment (EdisonLeeeee) [reply torch_geometric/utils/coalesce.py:43]: I'd prefer PyTorch-style code examples like that
https://github.com/pytorch/pytorch/blob/bfc6db0a5af1acf2a4cb864c334f17bcd08fc079/torch/nn/functional.py#L4375-L4390

Comment (rusty1s) [reply torch_geometric/utils/coalesce.py:43]: Got it, I guess this works for me as well.

File: torch_geometric/utils/coalesce.py [hunk 3]
```diff
@@ -37,6 +37,31 @@ def coalesce(
 
     :rtype: :class:`LongTensor` if :attr:`edge_attr` is :obj:`None`, else
         (:class:`LongTensor`, :obj:`Tensor` or :obj:`List[Tensor]]`)
+
+    Example:
```
Comment (rusty1s) [inline torch_geometric/utils/coalesce.py:41]:
```suggestion
    Examples::
```

File: torch_geometric/utils/convert.py [hunk 4]
```diff
@@ -19,6 +19,16 @@ def to_scipy_sparse_matrix(edge_index, edge_attr=None, num_nodes=None):
             edge features. (default: :obj:`None`)
         num_nodes (int, optional): The number of nodes, *i.e.*
             :obj:`max_val + 1` of :attr:`index`. (default: :obj:`None`)
+
+    Examples:
```
Comment (rusty1s) [inline torch_geometric/utils/convert.py:23]:
```suggestion
    Examples::
```

File: torch_geometric/utils/convert.py [hunk 5]
```diff
@@ -39,6 +49,20 @@ def from_scipy_sparse_matrix(A):
 
     Args:
         A (scipy.sparse): A sparse matrix.
+
+    Examples:
```
Comment (rusty1s) [inline torch_geometric/utils/convert.py:53]:
```suggestion
    Examples::
```

File: torch_geometric/utils/convert.py [hunk 6]
```diff
@@ -39,6 +49,20 @@ def from_scipy_sparse_matrix(A):
...
+
+        >>> edge_index = torch.tensor([
+        ...     [0, 1, 1, 2, 2, 3],
+        ...     [1, 0, 2, 1, 3, 2],
+        ... ])
+        >>> adj = to_scipy_sparse_matrix(edge_index)
+
+        >>> `edge_index` and `edge_weight` are both returned
```
Comment (rusty1s) [inline torch_geometric/utils/convert.py:61]:
```suggestion
        >>> # `edge_index` and `edge_weight` are both returned
```

File: torch_geometric/utils/convert.py [hunk 7]
```diff
@@ -140,6 +175,19 @@ def from_networkx(G, group_node_attrs: Optional[Union[List[str], all]] = None,
...
+    Examples:
+
+        >>> edge_index = torch.tensor([
+        ...     [0, 1, 1, 2, 2, 3],
+        ...     [1, 0, 2, 1, 3, 2],
+        ... ])
+        >>> data = Data(edge_index=edge_index, num_nodes=4)
+        >>> g = to_networkx(data)
+
+        >>> A `Data` object is returned
```
Comment (rusty1s) [inline torch_geometric/utils/convert.py:188]:
```suggestion
        >>> # A `Data` object is returned
```

File: torch_geometric/utils/convert.py [hunk 8]
```diff
@@ -140,6 +175,19 @@ def from_networkx(G, group_node_attrs: Optional[Union[List[str], all]] = None,
 
         All :attr:`group_node_attrs` and :attr:`group_edge_attrs` values must
         be numeric.
+
+    Examples:
```
Comment (rusty1s) [inline torch_geometric/utils/convert.py:179]:
```suggestion
    Examples::
```

File: torch_geometric/utils/convert.py [hunk 9]
```diff
@@ -72,6 +96,17 @@ def to_networkx(data, node_attrs=None, edge_attrs=None, graph_attrs=None,
             :obj:`False`)
         remove_self_loops (bool, optional): If set to :obj:`True`, will not
             include self loops in the resulting graph. (default: :obj:`False`)
+
+    Examples:
```
Comment (rusty1s) [inline torch_geometric/utils/convert.py:100]:
```suggestion
    Examples::
```

File: torch_geometric/utils/degree.py [hunk 10]
```diff
@@ -19,6 +19,12 @@ def degree(index: Tensor, num_nodes: Optional[int] = None,
             returned tensor.
 
     :rtype: :class:`Tensor`
+
+    Examples:
```
Comment (rusty1s) [inline torch_geometric/utils/degree.py:23]:
```suggestion
    Examples::
```

File: torch_geometric/utils/dropout.py [hunk 11]
```diff
@@ -35,6 +35,24 @@ def dropout_adj(
             :obj:`max_val + 1` of :attr:`edge_index`. (default: :obj:`None`)
         training (bool, optional): If set to :obj:`False`, this operation is a
             no-op. (default: :obj:`True`)
+
+    Examples:
```
Comment (rusty1s) [inline torch_geometric/utils/dropout.py:39]:
```suggestion
    Examples::
```

File: torch_geometric/utils/get_laplacian.py [hunk 12]
```diff
@@ -35,6 +35,22 @@ def get_laplacian(edge_index, edge_weight: Optional[torch.Tensor] = None,
             in case :obj:`edge_weight=None`. (default: :obj:`None`)
         num_nodes (int, optional): The number of nodes, *i.e.*
             :obj:`max_val + 1` of :attr:`edge_index`. (default: :obj:`None`)
+
+    Examples:
```
Comment (rusty1s) [inline torch_geometric/utils/get_laplacian.py:39]:
```suggestion
    Examples::
```

File: torch_geometric/utils/homophily.py [hunk 13]
```diff
@@ -63,6 +63,22 @@ def homophily(edge_index: Adj, y: Tensor, batch: OptTensor = None,
             either :obj:`"edge"` (first formula), :obj:`"node"` (second
             formula) or :obj:`"edge_insensitive"` (third formula).
             (default: :obj:`"edge"`)
+
+    Examples:
```
Comment (rusty1s) [inline torch_geometric/utils/homophily.py:67]:
```suggestion
    Examples::
```

File: torch_geometric/utils/homophily.py [hunk 14]
```diff
@@ -63,6 +63,22 @@ def homophily(edge_index: Adj, y: Tensor, batch: OptTensor = None,
...
+
+        >>> edge_index = torch.tensor([[0, 1, 2, 3],
+        ...                            [1, 2, 0, 4]])
+        >>> y = torch.tensor([0, 0, 0, 0, 1])
+        >>> batch = torch.tensor([0, 0, 0, 1, 1])
```
Comment (rusty1s) [inline torch_geometric/utils/homophily.py:72]: not used?

File: torch_geometric/utils/isolated.py [hunk 15]
```diff
@@ -15,6 +15,16 @@ def contains_isolated_nodes(edge_index, num_nodes=None):
             :obj:`max_val + 1` of :attr:`edge_index`. (default: :obj:`None`)
 
     :rtype: bool
+
+    Examples:
```
Comment (rusty1s) [inline torch_geometric/utils/isolated.py:19]:
```suggestion
    Examples::
```

File: torch_geometric/utils/isolated.py [hunk 16]
```diff
@@ -36,6 +46,19 @@ def remove_isolated_nodes(edge_index, edge_attr=None, num_nodes=None):
             :obj:`max_val + 1` of :attr:`edge_index`. (default: :obj:`None`)
 
     :rtype: (LongTensor, Tensor, BoolTensor)
+
+    Examples:
```
Comment (rusty1s) [inline torch_geometric/utils/isolated.py:50]:
```suggestion
    Examples::
```

**Review by rusty1s — APPROVED** [review]
Thank you!

File: torch_geometric/utils/homophily.py (hunk 14 above)
Comment (EdisonLeeeee) [reply torch_geometric/utils/homophily.py:72]: My bad. Will soon be removed.

File: torch_geometric/utils/coalesce.py (hunk 3 above)
Comment (EdisonLeeeee) [reply torch_geometric/utils/coalesce.py:41]: I found `Examples::` and `Examples:` are quite different in doc after complied, where the latter is boldfaced.
+ `Examples::`
image

+ `Examples:`
image


Which one would you prefer?

Comment (rusty1s) [reply torch_geometric/utils/coalesce.py:41]: Interesting. Looks like we need to modify the CSS for the `::` syntax. Let's keep it as `Examples:` then :(

Comment (EdisonLeeeee) [reply torch_geometric/utils/coalesce.py:41]: Got it.

**Comment by EdisonLeeeee** [comment]
Jobs done @rusty1s.
